base/applications/notepad/dialog.c

```diff
@@ -942,7 +942,35 @@ VOID DIALOG_StatusBarUpdateCaretPos(VOID)
     col = ((ich < 0) ? 0 : (dwStart - ich));
 
     StringCchPrintf(buff, _countof(buff), Globals.szStatusBarLineCol, line + 1, col + 1);
-    SendMessage(Globals.hStatusBar, SB_SETTEXT, SBPART_CURPOS, (LPARAM)buff);
+
+    /* Get all text from the edit control to count words */
+    INT cchTotalText = GetWindowTextLengthW(Globals.hEdit);
```

Comment:
I'm sorry but this looks lie super unefficient, if that `DIALOG_StatusBarUpdateCaretPos` function is called at (almost) every key stroke, when the user enters some text.

You will have to rework your algorithm.
For example, you do the calculation of the initial number of words at the same time the text document is loaded ("nInitialWordCount"), then you recalculate it by looking (in this `DIALOG_StatusBarUpdateCaretPos`) whether the user is on an existing word and he types a separator while being within a word (then: word count increases), if he's typing multiple separators (no more increments of word count), if he's deleting separators (if two separated words are now joined together, the word count decreases), etc.